scripts/update-lockfiles.py

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env python3
+"""
+Script to update pixi.lock files in test directories.
+
+This script:
+1. Recursively searches for pixi.lock files in tests/data/pixi_build (or specified directory)
+2. Runs `pixi install` in each directory containing a pixi.lock file
+3. Stops execution on the first error encountered
+4. Accepts optional command-line argument to specify a specific subdirectory to process
+"""
+
+import argparse
+import subprocess
+import sys
+from pathlib import Path
+
+
+class PixiInstallError(Exception):
+    """Raised when pixi install fails."""
+    pass
+
+
+def run_command(
+    cmd: list[str], cwd: Path | None = None, capture_output: bool = True
+) -> tuple[int, str, str]:
+    """Run a command and return exit code, stdout, and stderr."""
+    result = subprocess.run(cmd, cwd=cwd, capture_output=capture_output, text=True)
+    return result.returncode, result.stdout, result.stderr
+
+
+def pixi_install(directory: Path) -> None:
+    """Run pixi install in the specified directory."""
+    print(f"🔄 Running pixi install in {directory}")
+    returncode, stdout, stderr = run_command(["pixi", "install"], cwd=directory)
```

Comment:
Running `pixi lock` makes a bit more sense here, I feel :)

Reply:
but I agree that lock seems more ok in this case

Reply:
I don't think so. Our testsuite does that for us